spec/features/consumer/shopping/orders_spec.rb

```diff
@@ -3,6 +3,79 @@
 feature "Order Management", js: true do
   include AuthenticationWorkflow
 
+  describe "viewing a completed order" do
+    let!(:distributor) { create(:distributor_enterprise) }
+    let!(:customer) { create(:customer, user: user, enterprise: distributor) }
+    let!(:order_cycle) { create(:simple_order_cycle, distributors: [distributor]) }
+
+    let!(:bill_address) { create(:address) }
+    let!(:ship_address) { create(:address) }
+    let!(:shipping_method) { create(:free_shipping_method, distributors: [distributor]) }
+
+    let!(:order) do
+      create(:order_with_credit_payment,
+        customer: customer,
+        user: user,
+        distributor: distributor,
+        order_cycle: order_cycle
+      )
+    end
+
+    before do
+      # For some reason, both bill_address and ship_address are not set
+      # automatically.
+      #
+      # Also, assigning the shipping_method to a ShippingMethod instance results
+      # in a SystemStackError.
+      order.update_attributes!(
+        bill_address: bill_address,
+        ship_address: ship_address,
+        shipping_method_id: shipping_method.id
+      )
```

Comment:
why is this not done above like:

let!(:order) do
      create(:order_with_credit_payment,
        customer: customer,
        user: user,
        distributor: distributor,
        order_cycle: order_cycle
bill_address: bill_address,
        ship_address: ship_address,
        shipping_method_id: shipping_method.id
      )
    end

Reply:
@luisramos0 The current state of `Spree::Order` and how the factories are set up it turns out do not allow that. Please see thread here: https://openfoodnetwork.slack.com/messages/C2GQ45KNU/convo/C2GQ45KNU-1536159429.000100/ That leads to `SystemStackError` `stack level too deep`.

Reply:
ok, thanks. just asking, and learning :-)